### PR TITLE
Ovl_En_Minifrog OK with partial documentation

### PR DESCRIPTION
**Pull request by engineer124**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

More frogs! Hip Hip Hurrrrrribbit! All functions were named, but some were quite difficult to choose a good name, so I am quite open to better suggestions. 

En_Minifrog gv

### CODE REVIEW

File: include/functions.h [hunk 1]
```diff
@@ -795,7 +796,7 @@ f32* func_800B8248(PosRot* param_1, ActorPlayer* param_2);
 // void func_800B83BC(void);
 // void func_800B83F8(void);
 s32 func_800B84D0(Actor* actor, GlobalContext* globalCtx);
-// UNK_TYPE4 func_800B8500(Actor* actor, GlobalContext* globalCtx, f32 fParm3, f32 fParm4, s32 param_5);
+UNK_TYPE4 func_800B8500(Actor* actor, GlobalContext* globalCtx, f32 fParm3, f32 fParm4, s32 param_5);
```
Comment (jpburnett) [inline include/functions.h:799]: This looks to be a void return type as well, unless it needs the UNK_TYPE4?

**Review by jpburnett — APPROVED** [review]
Toad-ally awesome work. Just a thought on one part.

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c [hunk 2]
```diff
@@ -21,60 +43,602 @@ const ActorInit En_Minifrog_InitVars = {
...
+static u16 isFrogReturnedFlags[] = {
+    (0x00 << 8) | 0x00, 
+    (0x20 << 8) | 0x40, 
+    (0x20 << 8) | 0x80, 
+    (0x21 << 8) | 0x01, 
+    (0x21 << 8) | 0x02,
+};
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_En_Minifrog_0x808A3670/func_808A4AC8.asm")
+static s32 D_808A4D88 = false;
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c:85]:
```suggestion
static s32 isInitialized = false;
```

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c [hunk 3]
```diff
@@ -21,60 +43,602 @@ const ActorInit En_Minifrog_InitVars = {
...
+    }
+
+    this->actor.speedXZ = 0.0f;
+    this->actionFunc = EnMinifrog_Idle;
+    this->jumpState = MINIFROG_STATE_GROUND;
+    this->flags = 0;
+    this->timer = 0;
+
+    if (1) {}
+    if (!((this->actor.params & 0xF0) >> 4)) { // Frogs scattered throughout the world
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c:120]: Needs a macro for params access.

**Review by Kenix3 — CHANGES_REQUESTED** [review]

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c (hunk 3 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c]: Done! Hopefully it's in the right style, I haven't done a macro like that before

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.h [hunk 4]
```diff
@@ -5,9 +5,37 @@
 
 struct EnMinifrog;
 
+typedef void (*EnMinifrogActionFunc)(struct EnMinifrog*, GlobalContext*);
+
+#define EN_MINIFROG_IS_RETURNED(params) (((params)&0xF0) >> 4)
```
Comment (Kenix3) [inline src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.h:10]: Paramus takes an actor.

**Review by Kenix3 — CHANGES_REQUESTED** [review]
One change needed, otherwise looks good!

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c [hunk 5]
```diff
@@ -21,60 +43,602 @@ const ActorInit En_Minifrog_InitVars = {
...
+            ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+        }
+    }
+}
+
+// This can be seen when the Cyan and Pink frogs spawn after their respective minibosses
+void EnMinifrog_SpawnGrowAndShrink(EnMinifrog* this, GlobalContext* globalCtx) {
+    EnMinifrog_Jump(this);
+    if (this->timer > 0) {
+        Actor_SetScale(&this->actor, (0.01f + 0.0003f * this->timer * Math_SinS(0x1000 * (this->timer & 7))));
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c:318]: Would this match?
```suggestion
        Actor_SetScale(&this->actor, (0.01f + 0.0003f * this->timer * Math_SinS(0x1000 * (this->timer % 8))));
```

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c [hunk 6]
```diff
@@ -21,60 +43,602 @@ const ActorInit En_Minifrog_InitVars = {
...
+void EnMinifrog_Idle(EnMinifrog* this, GlobalContext* globalCtx) {
+    EnMinifrog_TurnToPlayer(this);
+    EnMinifrog_Jump(this);
+    EnMinifrog_JumpTimer(this);
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        this->actionFunc = EnMinifrog_ReturnFrogCutscene;
+        if (this->actor.cutscene != -1) {
+            this->flags |= 1;
+        }
+    } else if ((this->actor.xzDistToPlayer < 100.0f) && (Actor_IsLinkFacingActor(&this->actor, 0x3000, globalCtx) != 0) && (func_8012403C(globalCtx) == 0xD)) {
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c:335]: Would this match?
```suggestion
    } else if ((this->actor.xzDistToPlayer < 100.0f) && Actor_IsLinkFacingActor(&this->actor, 0x3000, globalCtx) && (func_8012403C(globalCtx) == 0xD)) {
```

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c [hunk 7]
```diff
@@ -21,60 +43,602 @@ const ActorInit En_Minifrog_InitVars = {
...
+        func_800B8614(&this->actor, globalCtx, 110.0f);
+    }
+}
+
+void EnMinifrog_SetupNextFrogInit(EnMinifrog* this, GlobalContext* globalCtx) {
+    struct EnMinifrog* nextFrog;
+    struct EnMinifrog* missingFrog;
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c:342]:
```suggestion
    EnMinifrog* nextFrog;
    EnMinifrog* missingFrog;
```

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c [hunk 8]
```diff
@@ -21,60 +43,602 @@ const ActorInit En_Minifrog_InitVars = {
...
+        ActorCutscene_Start(0x7C, NULL);
+        EnMinifrog_CheckChoirSuccess(this, globalCtx);
+        return;
+    } else if (this->actor.cutscene != -1 && ActorCutscene_GetCurrentIndex() == this->actor.cutscene) {
+        ActorCutscene_Stop(this->actor.cutscene);
+        ActorCutscene_SetIntentToPlay(0x7C);
+        return;
+    } else {
+        ActorCutscene_SetIntentToPlay(0x7C);
+    }
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c:391]:
```suggestion
    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
        EnMinifrog_CheckChoirSuccess(this, globalCtx);
    } else if (ActorCutscene_GetCanPlayNext(0x7C)) {
        ActorCutscene_Start(0x7C, NULL);
        EnMinifrog_CheckChoirSuccess(this, globalCtx);
    } else if (this->actor.cutscene != -1 && ActorCutscene_GetCurrentIndex() == this->actor.cutscene) {
        ActorCutscene_Stop(this->actor.cutscene);
        ActorCutscene_SetIntentToPlay(0x7C);
    } else {
        ActorCutscene_SetIntentToPlay(0x7C);
    }
```

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c [hunk 9]
```diff
@@ -21,60 +43,602 @@ const ActorInit En_Minifrog_InitVars = {
...
+        }
+    }
+
+    if (this->frog != NULL) {
+        this->actor.home.rot.z = 0;
+        this->actionFunc = EnMinifrog_NextFrogMissing;
+        this->timer = 60;
+        this->actor.home.rot.y = Actor_YawBetweenActors(&this->actor, &this->frog->actor);
+        func_801A1F88();
+        this->flags &= ~(0x100);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c:440]:
```suggestion
        this->flags &= ~0x100;
```

**Review by AngheloAlf — COMMENTED** [review]
Good work.
A few minor suggestions.

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c (hunk 5 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c:318]: It does not :(

File: src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c (hunk 8 above)
Comment (engineer124) [reply src/overlays/actors/ovl_En_Minifrog/z_en_minifrog.c:391]: I mentioned in a comment at the start of the function that the returns are required to match and I can't figure out why or how to match without them. I'll move the comment to `necessary to match` in front of the returns

**Pull request merged**